File: frontend/src/commands/environment-group.ts
import { invoke } from '@tauri-apps/api/core'

export interface EnvironmentGroup {
    name: string;
    description?: string;
}

export const environment_group_query_id = async (id: number): Promise<any> => {
    return await invoke('environment_group_query_id', { id })
};

export const environment_group_query = async (pageNum: number, pageSize: number): Promise<any> => {
    return await invoke('environment_group_query', { pageNum, pageSize })
};

export const environment_group_create = async (payload: EnvironmentGroup): Promise<any> => {
    return await invoke('environment_group_create', { payload })
};

export const environment_group_modify = async (id: number, payload: EnvironmentGroup): Promise<any> => {
    return await invoke('environment_group_modify', { id, payload })
};

export const environment_group_delete = async (id: number): Promise<any> => {
    return await invoke('environment_group_delete', { id })
};

File: crates/lp-views/src/command/environment_group.rs
use lp_cores::requests::JsonRespnse;

#[tauri::command]
pub async fn environment_group_query_id(id: u32) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_group::query_by_id(id).await?)
}

#[tauri::command]
pub async fn environment_group_query(
    page_num: u32,
    page_size: u32,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_group::query(page_num, page_size).await?)
}

#[tauri::command]
pub async fn environment_group_create(
    name: String,
    description: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_group::create(&name, Some(description)).await?)
}

#[tauri::command]
pub async fn environment_group_modify(
    id: u32,
    name: String,
    description: String,
) -> Result<JsonRespnse, tauri::Error> {
    Ok(
        lp_services_remote::requests::environment_group::modify(id, &name, Some(description))
            .await?,
    )
}

#[tauri::command]
pub async fn environment_group_delete(id: u32) -> Result<JsonRespnse, tauri::Error> {
    Ok(lp_services_remote::requests::environment_group::delete(id).await?)
}
